src/App.tsx

```diff
@@ -2,45 +2,69 @@ import React from 'react';
 import { Navigate, Route, Routes } from 'react-router-dom';
 
 import { ProcessWrapperWrapper } from 'src/components/wrappers/ProcessWrapper';
+import { useApplicationMetadata } from 'src/features/applicationMetadata/ApplicationMetadataProvider';
 import { Entrypoint } from 'src/features/entrypoint/Entrypoint';
 import { InstanceProvider } from 'src/features/instance/InstanceContext';
+import { InstantiateContainer } from 'src/features/instantiate/containers/InstantiateContainer';
 import { PartySelection } from 'src/features/instantiate/containers/PartySelection';
 import { InstanceSelectionWrapper } from 'src/features/instantiate/selection/InstanceSelection';
+import type { ShowTypes } from 'src/features/applicationMetadata';
+export const DefaultComponent = () => {
+  const applicationMetadata = useApplicationMetadata();
+  const show: ShowTypes = applicationMetadata.onEntry?.show ?? 'new-instance';
+
+  if (show === 'select-instance') {
+    return (
+      <Navigate
+        to={'/instance-selection/'}
+        replace={true}
+      />
+    );
+  }
+
+  return <InstantiateContainer />;
+};
```

Comment:
Hmm, I'm not quite sure why this was needed? It looks like this tries to do the same as Entrypoint does (but entrypoint supports way more, such as stateless apps (i.e. apps where the root path should load a form).